packages/@aws-cdk/aws-logs/lib/log-retention.ts

```diff
@@ -39,6 +40,27 @@ export interface LogRetentionProps {
    * @default - AWS SDK default retry options
    */
   readonly logRetentionRetryOptions?: LogRetentionRetryOptions;
+
+  /**
+   * The deletionPolicy the LogRetention uses on log
+   * @default LogDeletionPolicy.RETAINLOG
+   */
+  readonly logDeletionPolicy?: LogDeletionPolicy;
```

Comment:
I think this is the wrong name for this. This should be clear that this the deletion policy for when the stack is deleted, otherwise it just sounds redundant to the `retention` field. Maybe something here like 'onStackDeletion`? I don't actually like that a lot either but maybe you have a better idea?

Reply:
The policy name for removing other resource when the stack is deleted is RemovalPolicy (https://github.com/aws/aws-cdk/search?q=RemovalPolicy), so maybe it is okay to just call it logDeletionPolicy?

Reply:
Let's just call it `removalPolicy` to be consistent with the other naming conventions.

Reply:
Still needs updating. In fact, there's no reason to not just use the core [RemovalPolicy](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/core/lib/removal-policy.ts) as the type here. Let's not add duplicate code when the type already exists.